src/lib/hash.ts

```diff
@@ -39,3 +43,25 @@ const Poseidon = {
 
   Sponge,
 };
+
+const prefixes = Poseidon_.prefixes;
+
+function salt(prefix: string) {
+  return Poseidon.update(Poseidon.initialState, [prefixToField(prefix)]);
+}
+
+// same as Random_oracle.prefix_to_field in OCaml
```

Comment:
Why don't we use the OCaml binding of this function instead of defining our own version here?

Reply:
I could've done that, yeah. I guess I just try to slowly move things away from OCaml and to JS when I can. Makes better error messages and easier to look at the code for us and outside contributors. And maybe, eventually, smaller size if we can get rid of some big chunk of OCaml dependencies, like mina_base for example

Reply:
Sounds like a good idea to me :D